source/tutorials/nix-language.md

```diff
@@ -0,0 +1,799 @@
+
+# File system paths
+
+Nix language offers additional convenience for file system paths.[^3]
+
+Absolute paths always start with a slash (`/`):
+
+    /absolute/path
+
+Paths are relative when they contain at least one slash (`/`) but to not start with one.
+They are relative to the file containing the expression:
+
+    ./relative
+
+    relative/path
+
+[^3]: Details: [Nix manual - primitive data types][manual-primitives]
+
+## Search path
+
+Also known as “angle bracket syntax”.
+
+    <nixpkgs>
+
+The value of a named path is a file system path that depends on the contents of the [`$NIX_PATH`][NIX_PATH] environment variable.
+
+In practice, `<nixpkgs>` points to the file system path of some revision of the [Nix package collection][nixpkgs].
+For example, `<nixpkgs/lib>` points to the subdirectory `lib` of that file system path.
+
+[NIX_PATH]: https://nixos.org/manual/nix/unstable/command-ref/env-common.html?highlight=nix_path#env-NIX_PATH
+[nixpkgs]: https://github.org/NixOS/nixpkgs
+[manual-primitives]: https://nixos.org/manual/nix/stable/expressions/language-values.html#primitives
+
+# Character strings
+
+<!-- TODO: introduction -->
+
+## String interpolation
+
+<!-- TODO: details -->
+
+```nix
+let
+  name = "Nix";
+in
+"hello ${name}"
+```
+
+    "hello Nix"
+
+## Indented strings
+
+    ''
+      multi
+      line
+      string
+    ''
+
+You will recognize indented strings by *double single quotes*.
+Equal amounts of prepended white space are trimmed from the result.
+
+Example:
+
+```nix
+''
+  one
+   two
+    three
+''
+```
+
+    "one\n two\n  three\n"
+
+See [escaping rules]().
+
+# Using existing functions
+
+<!-- TODO: built-ins and library -->
+
+# Building software using side effects
+
+So far we have only covered what we call *pure expressions*:
+declaring data and transforming it with functions.
+
+Building software requires interaction with the outside world, which we call *side effects*.
+
+There are two main side effects in the Nix language that are relevant here:
+1. Reading files from the file system as build inputs
+2. Writing files to the file system as build outputs
+
+<!-- TODO: side effects - fetchers and derivations -->
+
+See Nix Pills for a detailed explanation on how Nix the package manager builds software using library functions and packages from the Nix package collection.
+
+# Summary
+
+You should now be able to read Nix language code for simple packages and configurations, and come up with similiar explanations of the following examples.
+
+Example:
+
+```nix
+{ pkgs ? import <nixpkgs> {}, ... }:
+pkgs.mkShell {
+  # ...
+}
+```
+
+Explanation:
+
+This expression is a function that takes an attribute set as an argument.
+If the argument has the attribute `pkgs`, it will be used in the function body.
+Otherwise, the default value of importing from the search path `<nixpkgs>` and calling the resulting function with an empty attribute set will be used.
```

Comment:
⚠️ You didn't explain `import` yet, what it does, and how to use it

Reply:
There will be a section on import, this PR is still a draft.